src/nvim/hardcopy.c

```diff
@@ -3080,16 +3079,18 @@ void mch_print_set_font(int iBold, int iItalic, int iUnderline)
   }
 }
 
-void mch_print_set_bg(long_u bgcol)
+void mch_print_set_bg(uint32_t bgcol)
 {
+  assert(bgcol <= INT_MAX);
   prt_bgcol = (int)bgcol;
   prt_attribute_change = TRUE;
   prt_need_bgcol = TRUE;
 }
 
-void mch_print_set_fg(long_u fgcol)
+void mch_print_set_fg(uint32_t fgcol)
 {
-  if (fgcol != (long_u)prt_fgcol) {
+  assert(fgcol <= INT_MAX);
+  if ((int)fgcol != prt_fgcol) {
```

Comment:
We could change the type of `prt_fgcol` from `int` to `uint32_t` and drop all casts and range checks.